packages/angular/cli/src/commands/completion/cli.ts

```diff
@@ -8,18 +8,57 @@
 
 import { join } from 'path';
 import yargs, { Argv } from 'yargs';
-import { CommandModule, CommandModuleImplementation } from '../../command-builder/command-module';
+import {
+  CommandModule,
+  CommandModuleImplementation,
+  CommandScope,
+} from '../../command-builder/command-module';
+import { addCommandModuleToYargs } from '../../command-builder/utilities/command';
+import { colors } from '../../utilities/color';
+import { initializeAutocomplete } from '../../utilities/completion';
 
 export class CompletionCommandModule extends CommandModule implements CommandModuleImplementation {
   command = 'completion';
   describe = 'Generate a bash and zsh real-time type-ahead autocompletion script.';
   longDescriptionPath = join(__dirname, 'long-description.md');
 
   builder(localYargs: Argv): Argv {
-    return localYargs;
+    return addCommandModuleToYargs(localYargs, CompletionInitCommandModule, this.context);
   }
 
   run(): void {
     yargs.showCompletionScript();
   }
 }
+
+class CompletionInitCommandModule extends CommandModule implements CommandModuleImplementation {
+  command = 'init';
```

Comment:
Maybe instead of having a sub command, the prompt should be in the `completion` command?

Reply:
I see `ng completion` as responsible for generating the shell configuration for auto completion, while `ng completion init` is for setting up your `.bashrc` to use `ng completion`. Since these are two different actions, I think they should be different commands.

Considering that `ng completion` is something that users won't generally be calling themselves, I could see changing `ng completion` to initialize the setup by appending `source <(ng completion --get-shell-config)` or something if we want to view that part as more of an implementation detail. I could go either way whether `--get-shell-config` should be a flag on `ng completion` or a subcommand.

Reply:
I like the `get-shell-config` concept. Maybe since users will not need to use this command/option it can also be hidden both from help and docs (aio).

Reply:
I think I like `ng completion` being the one that users run to set up autocompletion and the `ng completion some-flag-or-command` being the one that is added to shell configs, since users care less about that. I'm don't feel very strongly about `some-flag-or-command` given that it still needs to be public API anyways just for people who customize their `.bashrc` setups heavily. I think I still slightly prefer a subcommand rather than a flag since it has a completely different use case and different outputs which would be a lot to change with a single flag. As for the subcommand name `get-completion-script` is a little long and redundant, but it's also not something that users regularly type, so we don't need to overthink it too much.

I guess where I'm landing is:
* `ng completion` modifies your `.bashrc` to set up autocompletion for you.
* `ng completion get-completion-script` prints the Yargs setup.

Reply:
@clydin suggested `ng completion script`, which I like since we don't need to repeat "completion" (and the "get" isn't that valuable either).